sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/db/AbstractDeleteQueryBuilder.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.service.write.db;
+
+import io.stargate.bridge.grpc.Values;
+import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.bridge.proto.QueryOuterClass.Query;
+import io.stargate.bridge.proto.QueryOuterClass.Value;
+import io.stargate.sgv2.common.cql.builder.BuiltCondition;
+import io.stargate.sgv2.common.cql.builder.Predicate;
+import io.stargate.sgv2.common.cql.builder.QueryBuilder;
+import io.stargate.sgv2.common.cql.builder.Term;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentProperties;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Shared base class for all document query builder. */
+public abstract class AbstractDeleteQueryBuilder {
```

Comment:
Do we want to convert these to `@Value.Immutable` as well?

Reply:
Naah we can keep as is..